mysqld_exporter.go

```diff
@@ -124,40 +125,66 @@ func (e *Exporter) scrape(scrapes chan<- []string) {
 	// fetch slave status
 	rows, err = db.Query("SHOW SLAVE STATUS")
 	if err != nil {
-		log.Println("error running show slave query on database: ", err)
+		log.Println("error running show slave query on database:", err)
 		e.error.Set(1)
 		e.duration.Set(float64(time.Now().UnixNano()-now) / 1000000000)
 		return
 	}
 	defer rows.Close()
 
+	var slaveCols []string
+
+	slaveCols, err = rows.Columns()
+	if err != nil {
+		log.Println("error retrieving column list:", err)
+		e.error.Set(1)
+		e.duration.Set(float64(time.Now().UnixNano()-now) / 1000000000)
+		return
+	}
+
+	var slaveData = make([]sql.RawBytes, len(slaveCols))
+
+	scanArgs := make([]interface{}, len(slaveCols))
```

Comment:
I know what interface{} means ;)
I guess this response means I should have added a comment here.  The reason I do this is because rows.Scan() needs a pointer to an object rather than the object itself.  &slaveData would give a pointer to the slice of byteslices, when what we need is for each element of slaveData to be a pointer.  Creating a slice of interfaces, with each element being a pointer to the underlying sql.NullByte is a simple and fairly standard way of doing this.

Does that clarify it?